Market Offers chart modifications

deusmax opened the pull request:
<!-- 
- make yourself familiar with the CONTRIBUTING.md if you have not already (https://github.com/bisq-network/bisq/blob/master/CONTRIBUTING.md)
- make sure you follow our [coding style guidelines][https://github.com/bisq-network/style/issues)
- pick a descriptive title
- provide some meaningful PR description below
- create the PR
- in case you receive a "Change request" and/or a NACK, please react within 30 days. If not, we will close your PR and it can not be up for compensation.
- After addressing the change request, __please re-request a review!__ Otherwise we might miss your PR as we tend to only look at pull requests tagged with a "review required".
-->
## Intro
This is an attempt to simplify the Market Offers chart, to provide a better  (I think) visual "feeling". 
In particular, the scientific precision of the X-axis values is confusing to the user: 

For example, for EUR this:
![Bisq-MarketOfferBook-EURold-2020-11-09](https://user-images.githubusercontent.com/142019/98598945-e469ed80-22e3-11eb-9929-703b192189b5.png)
becomes:
![Bisq-MarketOfferBook-EURnew-2020-11-09](https://user-images.githubusercontent.com/142019/98599008-fb104480-22e3-11eb-8c79-576f895c15c0.png)

For XMR:
![Bisq-MarketOfferBook-default-XMR-2020-11-08](https://user-images.githubusercontent.com/142019/98600419-3449b400-22e6-11eb-9f8f-7877b79d03c3.png)
becomes:
![Bisq-MarketOfferBook-new-XMR-2020-11-08](https://user-images.githubusercontent.com/142019/98600531-5fcc9e80-22e6-11eb-97b5-9dafa3d0ac89.png)

For GBP:
![Bisq-MarketOfferBook-GBPold-2020-11-09](https://user-images.githubusercontent.com/142019/98600994-10d33900-22e7-11eb-937b-2e77292cd82e.png)
becomes:
![Bisq-MarketOfferBook-GBPnew-2020-11-09](https://user-images.githubusercontent.com/142019/98601027-1fb9eb80-22e7-11eb-9de4-d1b1a3b6fa69.png)

## The BSQ problem
The above works well (my opinion of course) for the remaining currencies (including USD, which I forgot to capture a pic - sorry), except for BSQ.
BSQ offers have some crazy values. The current charts implementation, filters-out values that have a factor 3 difference, the outliers are removed.
In the current PR, the filtering-out has been removed. I believe if there is a problem, the data should show it, not hide it. Hiding bad data creates a false sense of OKness. Better to see it:
Current BSQ chart:
![Bisq-MarketOfferBook-BSQold-2020-11-09](https://user-images.githubusercontent.com/142019/98604344-2ac34a80-22ec-11eb-8d22-703d5a88d547.png)
becomes:
![Bisq-MarketOfferBook-BSQnew-2020-11-09](https://user-images.githubusercontent.com/142019/98604387-3ca4ed80-22ec-11eb-90e2-0312ccc5cc83.png)

Looking at the Sell BTC (red) table, we see the last few values represent orders of magnitude jumps from the main core.
Putting back the filtering is easy, as is taking it off. I hope a discussion on how to best handle this issue follows. Is there one ongoing, already ?

Perhaps an outliers-toggle switch like in that other DAO chart on BSQs.

## The translations
The Y-axis label is "BTC Offer Volume". As a multi-lingual configuration is used on this chart, a translation web service was used to provide stock translations, which probably sometimes fail miserably. I don't know the Bisq policy on this, please advice.

deusmax committed:
Improve view of market offers chart

  * Both axes use autoRange, which results in better
    rounded values for the tick mark labels numbers.
  * Filter-out data outside the factor limits.
  * New constant chartDataFactor for factor limits.
  * Formatted tick label currency numbers:
    - Fiat: display with grouping separators
    - Cryptos: remove trailing zeros in fractional parts
  * Add CSS class axisy for y-axis formatting.

chimp1984 commented:
Thanks for those improvements!

> In the current PR, the filtering-out has been removed. I believe if there is a problem, the data should show it, not hide it. Hiding bad data creates a false sense of OKness.

It is not bad data it is just people putting up offer far away from market price. In other currencies where we get an external price feed from other exchanges we have a limit of the max. deviation (I guess 50%) but for those currencies where no external price is available (as BSQ) that is not the case. I think thats the reason why we see in BSQ such outliers and in other markets not (would be probably the case as well if we would not have that limit). I guess ppl set up such offer in the hope that someone make a mistake at some point and take it.... 

So to make the chart usable, I think we have to add the filter again to hide the outliers. I think a toggle is not needed as the display of the outliers included chart is pretty useless. But no strong opinion on that.

chimp1984 commented on core/src/main/resources/i18n/displayStrings_de.properties:
```diff
@@ -52,6 +52,7 @@ shared.P2P=P2P
 shared.oneOffer=Angebot
 shared.multipleOffers=Angebote
 shared.Offer=Angebot
+shared.offerVolumeCode={0} Angebotsvolumen
```
Any edit of the other language files will be reverted once updated from translators. For devs displayStrings.properties is the oinly file to edit, the rest is done via Transifex.

deusmax replied:
Hi @chimp1984, 
Great! I will revert  the changes to the language files. Makes it easier !

chimp1984 commented:
> The Y-axis label is "BTC Offer Volume". As a multi-lingual configuration is used on this chart, a translation web service was used to provide stock translations, which probably sometimes fail miserably. I don't know the Bisq policy on this, please advice.

We have translators who work with Transifex. Devs only edit the displayStrings.properties, changes in the other files will have not effect as they get reverted. If a string is not traanslated the english version is the fallback.

deusmax commented:
On the BSQ **bad data** issue.
Exactly as you say, it is not **bad data**. The chart is doing its job correctly. The problem should be solved at the data level, if possible. Doing so will also appeal to new users, who get discouraged or scared from crazy numbers. The numbers are still in the table.
One way to control it is to place a cap on the daily % price change up or down. Like 20 or 50 percent.
This is interesting, but perhaps off-topic here.

I'll add the filtering back-in.

chimp1984 commented:
> One way to control it is to place a cap on the daily % price change up or down. Like 20 or 50 percent.

Yes to apply a 50% diff from market price would be probably good. Though those would be only applied to new offers, and I assume those offers are out since long will stay there for longer...
For BSQ it could be applied as we have sufficient data to calculate the average, but we could not apply that generally to a currency where not external price feed is there, as it might be that there are just very few trades or none at all... But a problem to irrelevant to spend time on ;-)

chimp1984 reviewed (changes requested):
Beside the removed outlieer filter for BSQ it looks good to me.

deusmax commented:
Removed the stock translations and added back the filtering as suggested.
The filtering is generic not specific to BSQ, as before.
Here is the new look of BSQ market offer chart:
![Bisq-MarketOfferBook-BSQnew-2020-11-10](https://user-images.githubusercontent.com/142019/98717351-9ca39e80-2395-11eb-8a1f-4c445961f13c.png)

chimp1984 reviewed (dismissed):
utACK

deusmax commented:
Needs another review, as I pushed a minor edit - replacing the hard-coded data cutoff value with a constant variable.
I was unaware of the previous review at the time.

chimp1984 reviewed (approved):
utACK

chimp1984 commented:
> Needs another review, as I pushed a minor edit - replacing the hard-coded data cutoff value with a constant variable.
> I was unaware of the previous review at the time.

No problem... Thanks for the PR!

deusmax commented:
It seems the change from openjdk10 to openjdk12, made Travis complain. 
Travis fixed !!

deusmax commented:
@chimp1984 
Wouldn't it be nice to have this PR part of v1.5 ?
It would contribute another visual change to the version bump.

ripcurlx reviewed (approved):
utACK

@deusmax I'll cherry-pick this commit into the v1.5.0 release branch as it doesn't include any critical change and can be verified/tested quite quickly.

deusmax commented:
Great !
